src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +230,218 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    return Cookie.cookie(REFRESH_TOKEN, refreshToken)
+        .setMaxAge(ttlSeconds)
+        .setSecure(true)
+        .setPath("/authn")
+        .setHttpOnly(true)
+        .encode();
+  }
+
+  private String accessTokenCookie(String accessToken, String accessTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(accessTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    return Cookie.cookie(ACCESS_TOKEN, accessToken)
+      .setMaxAge(ttlSeconds)
+      .setSecure(true)
+      .setHttpOnly(true)
+      .encode();
+  }
+
+  private Response tokenResponse(JsonObject tokens) {
+    String accessToken = tokens.getString(ACCESS_TOKEN);
+    String refreshToken = tokens.getString(REFRESH_TOKEN);
+    String accessTokenExpiration = tokens.getString(ACCESS_TOKEN_EXPIRATION);
+    String refreshTokenExpiration = tokens.getString(REFRESH_TOKEN_EXPIRATION);
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key -- which is what we need
+    // here.
+    var body = new JsonObject()
+        .put(ACCESS_TOKEN_EXPIRATION, accessTokenExpiration)
+        .put(REFRESH_TOKEN_EXPIRATION, refreshTokenExpiration)
+        .toString();
+    return Response.status(201)
+        .header(SET_COOKIE, refreshTokenCookie(refreshToken, refreshTokenExpiration))
+        .header(SET_COOKIE, accessTokenCookie(accessToken, accessTokenExpiration))
+        .type(MediaType.APPLICATION_JSON)
+        .entity(body)
+        .build();
+  }
+
+  private Response logoutResponse() {
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key -- which is what we need
+    // here.
+    return Response.status(204)
+        .header(SET_COOKIE, REFRESH_TOKEN + "=; Path=/authn; Expires=" + COOKIE_EXPIRES_FOR_DELETE)
+        .header(SET_COOKIE, ACCESS_TOKEN + "=; Expires=" + COOKIE_EXPIRES_FOR_DELETE)
+        .build();
+  }
+
+  private Response tokenResponseLegacy(JsonObject fetchTokenFuture) {
```

Comment:
The variable `fetchTokenFuture` should be renamed because it's a `JsonObject` and not a `Future<JsonObject>`. How about `tokenJson`?

Reply:
Sounds good